### ISSUE TITLE
[Serve] Fastapi 0.109.0 is breaking the Serve proxy 

### ISSUE THREAD

**Issue by sihanwang41** [ray-project/ray#42711]
### What happened + What you expected to happen

pip install fastapi==0.109.0.

run

```
import ray
import requests
from fastapi import FastAPI
from ray import serve

app = FastAPI()


@serve.deployment
@serve.ingress(app)
class MyFastAPIDeployment:
    @app.get("/")
    def root(self):
        return "Hello, world!"


serve.run(MyFastAPIDeployment.bind(), route_prefix="/hello")
resp = requests.get("http://localhost:8000/hello")
# Got 404 response
assert resp.json() == "Hello, world!"
```

Upstream issue: https://github.com/tiangolo/fastapi/discussions/10948

### Versions / Dependencies

master

### Reproduction script

N/A

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by sihanwang41** [comment]
pin fastapi old version: https://github.com/ray-project/ray/pull/42682

**Comment by edoakes** [comment]
Hopefully we can unpin by 2.10 (if new fastapi version is out by then)

**Comment by edoakes** [comment]
@antoniomdk yep, looks like FastAPI fixed the starlette issue with 0.109.2! We'll unpin this dep for the Ray 2.10 release.
